src/com/owncloud/android/ui/activity/FileDisplayActivity.java

```diff
@@ -765,27 +766,39 @@ private void requestSimpleUpload(Intent data, int resultCode) {
         OCFile currentDir = getCurrentDir();
         String remotePath = (currentDir != null) ? currentDir.getRemotePath() : OCFile.ROOT_PATH;
 
-        if (selectedImageUri.toString().startsWith(UriUtils.URI_CONTENT_SCHEME)) {
-//            Cursor cursor = getContentResolver().query(Uri.parse(filePath), null, null, null, null);
-//            try {
-//                if (cursor != null && cursor.moveToFirst()) {
-//                    String displayName = cursor.getString(cursor.getColumnIndex(
-//                            OpenableColumns.DISPLAY_NAME));
-//                    Log_OC.v(TAG, "Display Name: " + displayName);
-//
-//                    displayName.replace(File.separatorChar, '_');
-//                    displayName.replace(File.pathSeparatorChar, '_');
-//                    remotePath += displayName + DisplayUtils.getComposedFileExtension(filePath);
-//
-//                }
-//                // and what happens in case of error?; wrong target name for the upload
-//            } finally {
-//                cursor.close();
-//            }
-            // Pending to be fixed
-            Toast.makeText(this, R.string.common_error_unknown, Toast.LENGTH_SHORT).show();
-            return;
+        if (selectedFileUri.toString().startsWith(UriUtils.URI_CONTENT_SCHEME)) {
+            Cursor cursor = getContentResolver().query(selectedFileUri, null, null, null, null);
+            try {
+                if (cursor != null && cursor.moveToFirst()) {
+                    String displayName = cursor.getString(cursor.getColumnIndex(
+                            OpenableColumns.DISPLAY_NAME));
+                    Log_OC.v(TAG, "Display Name: " + displayName);
+
+                    displayName = displayName.replace(File.separatorChar, '_');
+
+                    remotePath += displayName;
+
+                    // Check if extension is not included in file display name and add file extension
+                    int pos = displayName.lastIndexOf('.');
+                    if (pos < 0) {
+                        remotePath += DisplayUtils.getFileExtension(filePath);
+                    }
 
+                    // URi and remote path parameters
+                    Uri[] uris = new Uri[]{selectedFileUri};
+                    String[] remotePaths = new String[]{remotePath};;
+
+                    // Call to copy and then upload the selected file
+                    copyThenUpload(uris, remotePaths);
+
+                    cursor.close();
+                    return;
+
+                }
```

Comment:
The ```else``` case corresponding to the ```if``` block ending here should also lead to showing an error message via a ```Toast```. Let's compare it with the error messages shown in activity ```Uploader```